bdk_chain/src/keychain/keychain_tracker.rs

```diff
@@ -21,6 +23,8 @@ pub struct KeychainTracker<K, P, T = Transaction> {
     /// Index between script pubkeys to transaction outputs
     pub txout_index: KeychainTxOutIndex<K>,
     chain_graph: ChainGraph<P, T>,
+    /// UTXOs selected by transactions as inputs
+    selected: RefCell<HashSet<OutPoint>>,
```

Comment:
I used `RefCell` to solve an issue that came up in `test_keychain_tracker` on line `332` where the compiler was not telling me I can't have an immutable and mutable borrow to the `tracker` in the same scope.  That was because `full_utxos` will take an immutable borrow to the tracker and `mark_selected` will take a mutable borrow.  What do you think will have been the best way to solve the issue.